hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/LeafQueue.java

```diff
@@ -2352,17 +2352,17 @@ private void updateQueuePreemptionMetrics(RMContainer rmc) {
     final long usedMillis = rmc.getFinishTime() - rmc.getCreationTime();
     final long usedSeconds = usedMillis / DateUtils.MILLIS_PER_SECOND;
     Resource containerResource = rmc.getAllocatedResource();
-    metrics.preemptContainer();
+    usageTracker.getMetrics().preemptContainer();
     long mbSeconds = (containerResource.getMemorySize() * usedMillis)
         / DateUtils.MILLIS_PER_SECOND;
     long vcSeconds = (containerResource.getVirtualCores() * usedMillis)
         / DateUtils.MILLIS_PER_SECOND;
-    metrics.updatePreemptedMemoryMBSeconds(mbSeconds);
-    metrics.updatePreemptedVcoreSeconds(vcSeconds);
-    metrics.updatePreemptedResources(containerResource);
-    metrics.updatePreemptedSecondsForCustomResources(containerResource,
+    usageTracker.getMetrics().updatePreemptedMemoryMBSeconds(mbSeconds);
+    usageTracker.getMetrics().updatePreemptedVcoreSeconds(vcSeconds);
+    usageTracker.getMetrics().updatePreemptedResources(containerResource);
+    usageTracker.getMetrics().updatePreemptedSecondsForCustomResources(containerResource,
         usedSeconds);
-    metrics.updatePreemptedForCustomResources(containerResource);
+    usageTracker.getMetrics().updatePreemptedForCustomResources(containerResource);
```

Comment:
Wouldn't it be simpler if we'd introduce a local metrics = usageTracker.getMetrics(); variable? It would reduce the change, and also would be little bit faster (it doesn't matter much, but there is no need to call getMetrics this many times)

Reply:
Good point, fixed it.